x-pack/legacy/plugins/siem/public/components/top_n/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { timelineDefaults } from '../../store/timeline/defaults';
+import { TimelineModel } from '../../store/timeline/model';
+import { combineQueries } from '../timeline/helpers';
+import { useTimelineTypeContext } from '../timeline/timeline_context';
+
+import { TopN } from './top_n';
+
+/** The currently active timeline always has this Redux ID */
+export const ACTIVE_TIMELINE_REDUX_ID = 'timeline-1';
+
+const EMPTY_FILTERS: Filter[] = [];
+const EMPTY_QUERY: Query = { query: '', language: 'kuery' };
+
+const makeMapStateToProps = () => {
+  const getGlobalQuerySelector = inputsSelectors.globalQuerySelector();
+  const getGlobalFiltersQuerySelector = inputsSelectors.globalFiltersQuerySelector();
+  const getTimeline = timelineSelectors.getTimelineByIdSelector();
+  const getInputsTimeline = inputsSelectors.getTimelineSelector();
+  const getKqlQueryTimeline = timelineSelectors.getKqlFilterQuerySelector();
+
+  // The mapped Redux state provided to this component includes the global
+  // filters that appear at the top of most views in the app, and all the
+  // filters in the active timeline:
+  const mapStateToProps = (state: State) => {
+    const activeTimeline: TimelineModel =
+      getTimeline(state, ACTIVE_TIMELINE_REDUX_ID) ?? timelineDefaults;
+    const activeTimelineFilters = activeTimeline.filters ?? EMPTY_FILTERS;
+    const activeTimelineInput: inputsModel.InputsRange = getInputsTimeline(state);
+
+    return {
+      activeTimelineFilters,
+      activeTimelineFrom: activeTimelineInput.timerange.from,
+      activeTimelineKqlQueryExpression: getKqlQueryTimeline(state, ACTIVE_TIMELINE_REDUX_ID),
+      activeTimelineTo: activeTimelineInput.timerange.to,
+      dataProviders: activeTimeline.dataProviders,
+      globalQuery: getGlobalQuerySelector(state),
+      globalFilters: getGlobalFiltersQuerySelector(state),
+      kqlMode: activeTimeline.kqlMode,
+    };
+  };
+
+  return mapStateToProps;
+};
+
+const mapDispatchToProps = { setAbsoluteRangeDatePicker: dispatchSetAbsoluteRangeDatePicker };
+
+const connector = connect(makeMapStateToProps, mapDispatchToProps);
+
+interface OwnProps {
+  browserFields: BrowserFields;
+  field: string;
+  indexPattern: IIndexPattern;
+  toggleTopN: () => void;
+  onFilterAdded?: () => void;
+  value?: string[] | string | null;
+}
+type PropsFromRedux = ConnectedProps<typeof connector>;
+type Props = OwnProps & PropsFromRedux;
+
+const StatefulTopNComponent: React.FC<Props> = ({
+  activeTimelineFilters,
+  activeTimelineFrom,
+  activeTimelineKqlQueryExpression,
+  activeTimelineTo,
+  browserFields,
+  dataProviders,
+  field,
+  globalFilters = EMPTY_FILTERS,
+  globalQuery = EMPTY_QUERY,
+  indexPattern,
+  kqlMode,
+  onFilterAdded,
+  setAbsoluteRangeDatePicker,
+  toggleTopN,
+  value,
+}) => {
+  const kibana = useKibana();
+  const { documentType, id: timelineId } = useTimelineTypeContext();
+
+  return (
+    <GlobalTime>
```

Comment:
I am worried that the use of GlobalTime here, normally we can only be used on the top of everything, it can not be used inside of component because it will messed up the tracking of the query. I mean we can not have to `GlobalTime` inside of each other. Let's talk about this one.